value.go

```diff
@@ -5,7 +5,8 @@ type Value interface {
 	GetID() string
 	GetCreatedAt() int64
 	GetUpdatedAt() int64
-	GetRelationshipIDs() []string
+	GetRelationships() Relationships
```

Comment:
Why the type alias? I suspect this is going to end up w/ a lot of casting for weird reasons. Stuff like:

```golang
func (this *SomeObject) GetRelationships() Relationships {
    return Relationships{Relationship("table1"), Relationship("table2")}
}
```

As opposed to w/o it:
```golang
func GetRelationships() []string {
    return []string{"table1", "table2"}
}
```

Seems verbose, is all. Not a hill worth fighting on though, up to you.

Reply:
This is what the intended usage looks like.

```go
func (t *testStruct) GetRelationships() (r Relationships) {
	r.Append(t.UserID)
	r.Append(t.ContactID)
	r.Append(t.GroupID)
	r.Append(t.Tags...)
	return
}
```

Reply:
One additional note, the example you provided is slightly off. The proper implementation would look like this:

```go
func GetRelationships() [][]string {
	return [][]string{
		[]string{"table1"},
		[]string{"table2"},
	}
}
```

Reply:
Since `GetRelationshipIDs` looks to be only used internally, can we continue to support it? It's marked as "deprecated" which generally means "it still works, but please stop using it, it will go away in the future", but it looks like it's actually broken, in that it no longer does what it used to. Can we put the old functionality back in, but mark it as deprecated? Then we can remove it in a future release. That would help everyone continue to upgrade, and not get stuck on old versions.

From https://blog.golang.org/godoc:
> Sometimes a struct field, function, type, or even a whole package becomes redundant or unnecessary, but must be kept for compatibility with existing programs. To signal that an identifier should not be used, add a paragraph to its doc comment that begins with "Deprecated:" followed by some information about the deprecation. There are a few examples [in the standard library](https://golang.org/search?q=Deprecated:).

Reply:
I'm down with adding the deprecated note for the docs. I still feel very conflicted with leaving the old method. It feels like an easy place for bugs to occur in the future. Also, the backwards compatibility idea really applies to libraries which have reached version 1. 

I'm not writing off your suggestion yet. I just need some time to bake on the idea and see how I feel. Let me get back to you later today!